client/app/account/billing/main/payments/billing-payments.controller.js

```diff
@@ -1,132 +1,93 @@
-            .finally(() => {
-                $timeout(() => {
-                    this.loaders.payments = false;
-                });
-            });
+    this.loadPayments = ($config) => {
+
+        let request = OvhApiMe.Deposit().v7().query().sort($config.sort.property, $config.sort.dir > 0 ? "ASC" : "DESC");
+
+        _.filter($config.criteria, "property", "date").forEach((crit) => {
+            switch (crit.operator) {
+            case "is":
+                request = request.addFilter("date", "ge", crit.value);
+                request = request.addFilter("date", "le", moment(crit.value).add(1, "day").format("YYYY-MM-DD"));
+                break;
+            case "isAfter":
+                request = request.addFilter("date", "ge", crit.value);
+                break;
+            case "isBefore":
+                request = request.addFilter("date", "ge", crit.value);
+                break;
+            default:
+                break;
+            }
+        });
+
+        _.filter($config.criteria, "property", "amount.value").forEach((crit) => {
+            request = request.addFilter("amount.value", {
+                is: "eq",
+                smaller: "lt",
+                bigger: "gt"
+            }[crit.operator], crit.value);
+        });
+
+        _.filter($config.criteria, "property", "paymentInfo.paymentType").forEach((crit) => {
+            request = request.addFilter("paymentInfo.paymentType", {
+                is: "eq",
+                isNot: "ne"
+            }[crit.operator], crit.value);
+        });
+
+        return $q.all({
+            count: request.clone().execute().$promise,
+            deposit: request.clone().expand()
+                .offset($config.offset - 1)
+                .limit($config.pageSize)
+                .execute().$promise
+        }).then(({ count, deposit }) => {
+            this.payments = _.map(deposit, "value");
+            return { count, deposit };
+        }).then(({ count, deposit }) => ({
+            data: _.map(deposit, "value"),
+            meta: {
+                totalCount: count.length
+            }
+        }));
     };
 
     this.getDatasToExport = () => {
-        const datasToReturn = [[$translate.instant("payments_table_head_id"), $translate.instant("payments_table_head_date"), $translate.instant("payments_table_head_amount"), $translate.instant("payments_table_head_type")]];
-
-        return datasToReturn.concat(this.paginatedPayments.map((payment) => [payment.billId, $filter("date")(payment.date, "mediumDate"), payment.amount.text, this.getTranslatedPaiementType(payment)]));
+        const header = [
+            $translate.instant("payments_table_head_id"),
+            $translate.instant("payments_table_head_date"),
+            $translate.instant("payments_table_head_amount"),
+            $translate.instant("payments_table_head_type")
+        ];
+        const result = [header];
+        return result.concat(this.payments.map((payment) => [
```

Comment:
Can't `this.payments` be `null || undefined` here ?

Reply:
_.map would returns an empty array

Reply:
You're using `Array.prototype.map`, not `_.map` 😅

Reply:
dude please ```this.payments = _.map(deposit, "value");```

Reply:
Still talking about `this.payments.map`

Reply:
1. Can't this.payments be null || undefined here ?
2. no because it is initialized using this.payments = _.map(deposit, "value");
3. so [].map never fails